using-babelfish/uast-querying.md

```diff
@@ -4,53 +4,41 @@ The clients support querying over the returned UAST to filter for specific nodes
 
-* `InternalType` is converted to the element name
-* `Token`, if available, is converted to an attribute with `token` as keyword and the actual token as value
-* Every `Role` is converted to an attribute concatenating a `role` prefix and the role name in CamelCase.
+* `@type` is converted to the element name
+* `@token`, if available, is converted to an attribute with `@token` as keyword and the actual token as value
+* Every `Role` is converted to an attribute with `@role` as keyword and the role name as value.
 * Every `Property` is converted to an attribute with the property keyword as keyword and the property value as value
-* `StartPosition`, if available, is mapped to three attributes:
-  * A `startOffset` attribute, with the offset as value
-  * A `startLine` attribute, with the line as value
-  * A `startCol` attribute, with the column as value
-* `EndPosition`, if available, is mapped to three attributes:
-  * A `endOffset` attribute, with the offset as value
-  * A `endLine` attribute, with the line as value
-  * A `endCol` attribute, with the column as value
+* `@pos`, if available, is mapped to a node with a `start` and `end` properties, each containing a node of `Position` type which in turn will have the `offset`, `line` and `col` properties.
 
 Internally, these are mapped in the XML node in the following way:
 
 ```markup
 <{{InternalType}}
-    token='{{Token}}'
+    @token='{{Token}}'
     {{for role in Roles}}
-    role{{role}}
+    @role{{role}}
     {{for key, value in Properties}}
     {{key}}='{{value}}'
-    startOffset={{StartPosition.Offset}}
-    startLine={{StartPosition.Line}}
-    startCol={{StartPosition.Col}}
-    endOffset={{EndPosition.Offset}}
-    endLine={{EndPosition.Line}}
-    endCol={{EndPosition.Col}}>
+    @pos{{start, end}}
     {{Children}}
 </{{InternalType}}>
 ```
 
 This means that both language specific queries \(InternalType, Properties\) and language agnostic queries \(Roles\) can be done. For example:
 
 * All the numeric literals in Python: `//Num`
-* All the numeric literals in ANY language: `//*[@roleNumber and @roleLiteral]`
+* All the numeric literals in ANY language: `//*[@role='Number' and @role='Literal']`
 * All the integer literals in Python: `//*[@NumType='int']`
 
 The query language also allows some more complex queries:
 
-* All the elements in the tree that have either start or end offsets: `//*[@startOffset or @endOffset]`
-* All the simple identifiers: `//*[@roleIdentifier and not(@roleQualified)]`
-* All the simple identifiers that don't have any positioning: `//*[@roleIdentifier and not(@roleQualified) and not(@startOffset) and not(@endOffset)]`
-* All the arguments in function calls: `//*[@roleCall and @roleArgument]`
-* All the numeric literals in binary arithmetic operators: `//*[@roleBinary and @roleOperator and @roleArithmetic]//*[@roleNumber and @roleLiteral]`
+* All the simple identifiers: `//*[@role='Identifier' and not(@role='Qualified')]`
```

Comment:
Maybe mention Semantic option as well, since it's the default?

Reply:
Added both, thanks!